receiver/memcachedreceiver/scraper.go

```diff
@@ -16,91 +16,260 @@ package memcachedreceiver
+				if parsedV, ok := r.parseInt(k, v); ok {
+					r.addToIntMetric(network, attributes, parsedV)
+				}
 			case "get_hits":
-				addIntSum(ilm.Metrics(), metadata.M.MemcachedGetHits.Init, now, parseInt(v))
+				attributes.Insert(metadata.A.Operation, pdata.NewAttributeValueString("get"))
+				attributes.Insert(metadata.A.Type, pdata.NewAttributeValueString("hit"))
+				if parsedV, ok := r.parseInt(k, v); ok {
+					r.addToIntMetric(operationCount, attributes, parsedV)
+				}
 			case "get_misses":
-				addIntSum(ilm.Metrics(), metadata.M.MemcachedGetMisses.Init, now, parseInt(v))
+				attributes.Insert(metadata.A.Operation, pdata.NewAttributeValueString("get"))
+				attributes.Insert(metadata.A.Type, pdata.NewAttributeValueString("miss"))
+				if parsedV, ok := r.parseInt(k, v); ok {
+					r.addToIntMetric(operationCount, attributes, parsedV)
+				}
+			case "incr_hits":
+				attributes.Insert(metadata.A.Operation, pdata.NewAttributeValueString("increment"))
+				attributes.Insert(metadata.A.Type, pdata.NewAttributeValueString("hit"))
+				if parsedV, ok := r.parseInt(k, v); ok {
+					r.addToIntMetric(operationCount, attributes, parsedV)
+				}
+			case "incr_misses":
+				attributes.Insert(metadata.A.Operation, pdata.NewAttributeValueString("increment"))
+				attributes.Insert(metadata.A.Type, pdata.NewAttributeValueString("miss"))
+				if parsedV, ok := r.parseInt(k, v); ok {
+					r.addToIntMetric(operationCount, attributes, parsedV)
+				}
+			case "decr_hits":
+				attributes.Insert(metadata.A.Operation, pdata.NewAttributeValueString("decrement"))
+				attributes.Insert(metadata.A.Type, pdata.NewAttributeValueString("hit"))
+				if parsedV, ok := r.parseInt(k, v); ok {
+					r.addToIntMetric(operationCount, attributes, parsedV)
+				}
+			case "decr_misses":
+				attributes.Insert(metadata.A.Operation, pdata.NewAttributeValueString("decrement"))
+				attributes.Insert(metadata.A.Type, pdata.NewAttributeValueString("miss"))
+				if parsedV, ok := r.parseInt(k, v); ok {
+					r.addToIntMetric(operationCount, attributes, parsedV)
+				}
+			case "rusage_system":
+				attributes.Insert(metadata.A.State, pdata.NewAttributeValueString("system"))
+				if parsedV, ok := r.parseFloat(k, v); ok {
+					r.addToDoubleMetric(rUsage, attributes, parsedV)
+				}
+
+			case "rusage_user":
+				attributes.Insert(metadata.A.State, pdata.NewAttributeValueString("user"))
+				if parsedV, ok := r.parseFloat(k, v); ok {
+					r.addToDoubleMetric(rUsage, attributes, parsedV)
+				}
 			}
 		}
-	}
 
+		// Calculated Metrics
+		attributes := pdata.NewAttributeMap()
+		attributes.Insert(metadata.A.Operation, pdata.NewAttributeValueString("increment"))
+		parsedHit, okHit := r.parseInt("incr_hits", stats.Stats["incr_hits"])
+		parsedMiss, okMiss := r.parseInt("incr_misses", stats.Stats["incr_misses"])
+		if okHit && okMiss {
+			r.addToDoubleMetric(hitRatio, attributes, calculateHitRatio(float64(parsedHit), float64(parsedMiss)))
```

Comment:
Why parse as ints and pass as floats? They are always going to be ints, so might as well just pass them like that and have the `hitRatio` function deal with it.